aws-android-sdk-iot/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java

```diff
@@ -723,20 +725,44 @@ public void onFailure(Throwable exception) {
 
     @Test
     public void mqttWebSocket() throws Exception {
+        websocketConnectionTest(IAM);
+    }
 
+    @Test
+    public void testCustomAuthConnect() throws Exception {
+        websocketConnectionTest(CUSTOM_AUTH);
+    }
+
+    private void websocketConnectionTest(AWSIotMqttManager.AuthenticationMode authMode) throws InterruptedException {
         final ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus> statuses = new ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus>();
         final ArrayList<String> messages = new ArrayList<String>();
 
         AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-w-ws", Region.getRegion(Regions.US_EAST_1), endpointPrefix);
 
         mqttManager.setAutoReconnect(false);
-        // connect using WebSockets and IAM credentials
-        mqttManager.connect(credentialsProvider, new AWSIotMqttClientStatusCallback() {
-            @Override
-            public void onStatusChanged(AWSIotMqttClientStatus status, Throwable throwable) {
-                statuses.add(status);
-            }
-        });
+
+        if (authMode.equals(CUSTOM_AUTH)) {
```

Comment:
It's always better to do these like:
```
FOR_SURE_NON_NULL.equals(possiblyNull)
```
When `possiblyNull` _is_ `== null`, then that statement evaluates as false.

However, the statement `possiblyNull.equals(FOR_SURE_NON_NULL)` throws a `NullPointerException` in that same situation.

^^^ Same comment for every place you have an `enumValue.equals(Enum.CONSTANT)`

Reply:
That makes lot of sense. It will definitely save us a lot of NPEs. Updated to follow the pattern.